### ISSUE TITLE
An error was encountered while running hmmsearch.

### ISSUE THREAD

**Issue by RahiPraveen** [Ecogenomics/GTDBTk#507]
Hello there,

I was trying to run the identify step and encountering this error, several times.
[2023-04-13 14:12:26] INFO: GTDB-Tk v2.1.1
[2023-04-13 14:12:26] INFO: gtdbtk identify --genome_dir /pasteur/zeus/projets/p02/CRBIP_omics/Praveen_data/Genomes_bergeys --out_dir /pasteur/zeus/projets/p02/CRBIP_omics/Praveen_data/Genomes_bergeys/Identity --cpus 10
[2023-04-13 14:12:26] INFO: Using GTDB-Tk reference data version r207: /opt/gensoft/data/GTDBTk/2.1.1
[2023-04-13 14:12:26] INFO: Identifying markers in 84 genomes with 10 threads.
[2023-04-13 14:12:26] TASK: Running Prodigal V2.6.3 to identify genes.
[2023-04-13 14:17:42] INFO: Completed 84 genomes in 5.27 minutes (15.94 genomes/minute).
[2023-04-13 14:17:42] TASK: Identifying TIGRFAM protein families.
[2023-04-13 14:19:24] INFO: Completed 84 genomes in 1.70 minutes (49.38 genomes/minute).
[2023-04-13 14:19:24] TASK: Identifying Pfam protein families.
[2023-04-13 14:19:30] ERROR: An error was encountered while running hmmsearch.
[2023-04-13 14:19:30] ERROR: Controlled exit resulting from an unrecoverable error or warning.

**Comment by pchaumeil** [comment]
Hi,
Can you successfully run `gtdbtk test`?

**Comment by RahiPraveen** [comment]
Yes, it could run gtdbtk test, 
[2023-04-13 14:44:40] INFO: GTDB-Tk v2.1.1
[2023-04-13 14:44:40] INFO: gtdbtk test --out_dir /pasteur/zeus/projets/p02/CRBIP_omics/Praveen_data/Genomes_bergeys/out --cpus 3
[2023-04-13 14:44:40] INFO: Using GTDB-Tk reference data version r207: /opt/gensoft/data/GTDBTk/2.1.1
[2023-04-13 14:44:40] INFO: Command: gtdbtk classify_wf --genome_dir /pasteur/zeus/projets/p02/CRBIP_omics/Praveen_data/Genomes_bergeys/out/genomes --out_dir /pasteur/zeus/projets/p02/CRBIP_omics/Praveen_data/Genomes_bergeys/out/output --cpus 3 -f
[2023-04-13 14:51:10] INFO: Test has successfully finished.

**Comment by RahiPraveen** [comment]
Test execution log output:
[2023-04-13 14:44:41] INFO: GTDB-Tk v2.1.1
[2023-04-13 14:44:41] INFO: gtdbtk classify_wf --genome_dir /pasteur/zeus/projets/p02/CRBIP_omics/Praveen_data/Genomes_bergeys/out/genomes --out_dir /pasteur/zeus/projets/p02/CRBIP_omics/Praveen_data/Genomes_bergeys/out/output --cpus 3 -f
[2023-04-13 14:44:41] INFO: Using GTDB-Tk reference data version r207: /opt/gensoft/data/GTDBTk/2.1.1
[2023-04-13 14:44:41] INFO: Identifying markers in 3 genomes with 3 threads.
[2023-04-13 14:44:41] TASK: Running Prodigal V2.6.3 to identify genes.
[2023-04-13 14:44:59] INFO: Completed 3 genomes in 17.89 seconds (5.96 seconds/genome).
[2023-04-13 14:44:59] TASK: Identifying TIGRFAM protein families.
[2023-04-13 14:45:04] INFO: Completed 3 genomes in 5.15 seconds (1.72 seconds/genome).
[2023-04-13 14:45:04] TASK: Identifying Pfam protein families.
[2023-04-13 14:45:05] INFO: Completed 3 genomes in 0.34 seconds (8.73 genomes/second).
[2023-04-13 14:45:05] INFO: Annotations done using HMMER 3.3 (Nov 2019).
[2023-04-13 14:45:05] TASK: Summarising identified marker genes.
[2023-04-13 14:45:05] INFO: Completed 3 genomes in 0.07 seconds (42.01 genomes/second).
[2023-04-13 14:45:05] INFO: Done.
[2023-04-13 14:45:05] INFO: Aligning markers in 3 genomes with 3 CPUs.
[2023-04-13 14:45:05] INFO: Processing 3 genomes identified as archaeal.
[2023-04-13 14:45:05] INFO: Read concatenated alignment for 3,412 GTDB genomes.
[2023-04-13 14:45:05] TASK: Generating concatenated alignment for each marker.
[2023-04-13 14:45:05] INFO: Completed 3 genomes in 0.02 seconds (142.10 genomes/second).
[2023-04-13 14:45:05] TASK: Aligning 53 identified markers using hmmalign 3.3 (Nov 2019).
[2023-04-13 14:45:06] INFO: Completed 53 markers in 1.18 seconds (44.81 markers/second).
[2023-04-13 14:45:06] TASK: Masking columns of archaeal multiple sequence alignment using canonical mask.
[2023-04-13 14:45:13] INFO: Completed 3,415 sequences in 6.88 seconds (496.42 sequences/second).
[2023-04-13 14:45:13] INFO: Masked archaeal alignment from 13,540 to 10,153 AAs.
[2023-04-13 14:45:13] INFO: 0 archaeal user genomes have amino acids in <10.0% of columns in filtered MSA.
[2023-04-13 14:45:13] INFO: Creating concatenated alignment for 3,415 archaeal GTDB and user genomes.
[2023-04-13 14:45:16] INFO: Creating concatenated alignment for 3 archaeal user genomes.
[2023-04-13 14:45:16] INFO: Done.
[2023-04-13 14:45:17] TASK: Placing 3 archaeal genomes into reference tree with pplacer using 3 CPUs (be patient).
[2023-04-13 14:45:17] INFO: pplacer version: v1.1.alpha19-0-g807f6f3
[2023-04-13 14:50:59] INFO: Calculating RED values based on reference tree.
[2023-04-13 14:50:59] TASK: Traversing tree to determine classification method.
[2023-04-13 14:50:59] INFO: Completed 3 genomes in 0.00 seconds (4,036.87 genomes/second).
[2023-04-13 14:51:00] TASK: Calculating average nucleotide identity using FastANI (v1.33).
[2023-04-13 14:51:10] INFO: Completed 54 comparisons in 10.60 seconds (5.10 comparisons/second).
[2023-04-13 14:51:10] INFO: 3 genome(s) have been classified using FastANI and pplacer.
[2023-04-13 14:51:10] INFO: Note that Tk classification mode is insufficient for publication of new taxonomic designations. New designations should be based on one or more de novo trees, an example of which can be produced by Tk in de novo mode.
[2023-04-13 14:51:10] INFO: Done.
[2023-04-13 14:51:10] INFO: Removing intermediate files.
[2023-04-13 14:51:10] INFO: Intermediate files removed.
[2023-04-13 14:51:10] INFO: Done.

**Comment by pchaumeil** [comment]
Seems one of your 84 genomes may be causing the issue. do all the genomes have a good completeness?

**Comment by RahiPraveen** [comment]
Yes, I confirmed the completeness of all genomes, and all are more than 99% complete. 
I am also sharing checkM results
[bin_stats_ext.txt](https://github.com/Ecogenomics/GTDBTk/files/11232212/bin_stats_ext.txt)

**Comment by pchaumeil** [comment]
Can you provide one genome ( or a subset of genomes) causing this issue?

**Comment by pchaumeil** [comment]
These genomes have duplicates header after the prodigal step. 
As an example, for `Ko1_M1.fna`:

```
>> grep '>' Problematic_genomes/Ko1_M1.fna                                                                                          
>chromosome :: CP008841.1
>plasmid pKOXM1A :: CP008842.1
>plasmid pKOXM1B :: CP008843.1
>plasmid pKOXM1C :: CP008844.1
>plasmid pKOXM1D :: CP008845.1
```

Prodigal only takes the first part of the header and add a suffix to it, so when processing the first plasmid sequence (`>plasmid pKOXM1A :: CP008842.1`) you will get a list of potential genes : `>plasmid_1`,`>plasmid_2`,`>plasmid_3`....etc
The same will happen with the next plasmid sequence ( `>plasmid pKOXM1B :: CP008843.1`) which will generates the same duplicated potential genes ids ( `>plasmid_1`,`>plasmid_2`,`>plasmid_3` ).
Having duplicate headers will stop marker gene searches.

I would recommend making the headers unique in your genome files :
```
>plasmid pKOXM1A :: CP008842.1
>plasmid pKOXM1B :: CP008843.1
>plasmid pKOXM1C :: CP008844.1
>plasmid pKOXM1D :: CP008845.1
```

would become 
```
>plasmid_1 pKOXM1A :: CP008842.1
>plasmid_2 pKOXM1B :: CP008843.1
>plasmid_3 pKOXM1C :: CP008844.1
>plasmid_4 pKOXM1D :: CP008845.1
```

**Comment by RahiPraveen** [comment]
Hi @pchaumeil , It worked after the changes you suggested. 
Thank you very much :)
